app/poi_api/poi_service.py

```
@@ -0,0 +1,106 @@
+
+    def get_POI(self, poi_id):
+        try:
+            poi_ref = firestore_db.collection("POI")
+            poi = poi_ref.document(poi_id).get()
+            poi_dict = poi.to_dict()
+            if poi_dict is not None:
+                return self._create_poi_object(poi_dict)
+            else:
+                raise POINotFoundError(
+                    f"The POI with the id: {poi_id} could not be found", 404
+                )
+        except POINotFoundError as e:
+            raise POINotFoundError(
+                f"The POI with the id: {poi_id} could not be found", 404
+            )
+        except Exception as e:
+            raise Exception(f"An error occured: {e}", 404)
+
+    def suggest_new_POI(self, poi_suggestion):
+        try:
+            poi_suggestion_ref = firestore_db.collection("POI_proposal").document()
+            # Generate id for poi suggestion document
+            pid = poi_suggestion_ref.id
+            poi_suggestion_instance = self._create_POI_suggestion(pid, poi_suggestion)
+            self._save_POI_suggestion(poi_suggestion_instance, poi_suggestion_ref)
+            return {"success": True, "poi_suggestion_id": pid}
+        except Exception as e:
+            raise InvalidPOISuggestionError(f"Invalid POI submission: {e}", 440)
+
+    # TODO: Add User parameter
+    # TODO: Create user object for submitted_by
+    def _create_POI_suggestion(self, pid, poi_suggestion):
+        try:
+            suggestion_name = poi_suggestion.get("suggestion_name")
+            notes = poi_suggestion.get("notes")
+            submission_time = datetime.now()
+            submitted_by = poi_suggestion.get("submitted_by")
+            if suggestion_name is None or submitted_by is None:
+                raise Exception()
+            return POI_suggestions(
+                pid, suggestion_name, notes, submission_time, submitted_by
+            )
+        except Exception as e:
+            raise Exception(f"An error occured: {e}")

```

yeah these catch all exceptions need to go.